tests/phpunit/tests/passwordHash.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * Tests for the PasswordHash external library.
+ *
+ * @covers PasswordHash
+ */
+class Tests_PasswordHash extends WP_UnitTestCase {
+
+	public static function set_up_before_class() {
+		parent::set_up_before_class();
+
+		require_once ABSPATH . WPINC . '/class-phpass.php';
+	}
+
+	/**
+	 * Tests that PasswordHash::gensalt_blowfish() does not throw a deprecation
+	 * notice on PHP 8.1 for "Implicit conversion from float to int loses precision".
+	 *
+	 * @ticket 56340
+	 *
+	 * @covers PasswordHash::gensalt_blowfish
+	 */
+	public function test_gensalt_blowfish_should_not_throw_deprecation_notice_in_php81() {
+		if ( version_compare( PHP_VERSION, '8.1.0', '<' ) ) {
+			$this->markTestSkipped( 'This test should only run on PHP 8.1 or greater.' );
+		}
```

Comment:
If this would remain, it can be replaced by a `@requires PHP 8.1.0` annotation (please double-check the syntax, but you get my drift).

Reply:
Nice! After checking [the PHPUnit docs](https://phpunit.readthedocs.io/en/9.5/incomplete-and-skipped-tests.html#skipping-tests-using-requires), I've added `@requires PHP 8.1`, and tested that it runs on PHP 8.1 and 8.2 (not sure why there's a `>=` syntax available if `Requires PHP <version>` does it anyway).

Reply:
>  (not sure why there's a >= syntax available...

The comparison syntax is available to allow for `<` or `=`, i.e. tests which should only run on "less than" or "equal to" versions.
As the PHP native `version_compare()` function is used under the hood for the implementation, the `>=` syntax is automatically also supported, though in practice not needed as it's the default behaviour.

(and the only reason I know this is because i ran into a quirk with this a while back and updated the PHPUnit docs to expand the information about these operators).

Reply:
Ah, thanks a lot for the info!